app/controllers/consents_controller.rb

```diff
@@ -35,6 +35,15 @@ def create
   end
 
   def update
+    if consent_agree_params.present?
+      @draft_consent.assign_attributes(consent_agree_params)
+      if @draft_consent.save(context: :edit_consent)
+        redirect_to action: :edit_consent
+      else
+        render :edit_who
+      end
+    end
+
```

Comment:
Why have to `if consent_agree_params.present?` sections?

Reply:
Rebase mistake!

Reply:
Aahh, yes. This controller action really is giving me hives, though, with the checks multiple branches of logic that could be followed if more than one of these `consent_*_params.present?` come back true.

Not to suggest we do it in this PR, but how do you feel about splitting them into sub-controllers like [Vaccinations::BatchesController](https://github.com/nhsuk/record-childrens-vaccinations/blob/5a9cd92b3e67de6873196abb1558014b9b37251d/app/controllers/vaccinations/batches_controller.rb)?